frontend/src/app/modules/user/update-password-form/update-password-form.component.ts

```diff
@@ -0,0 +1,73 @@
+import {
+  Component, EventEmitter, OnInit, Output
+} from '@angular/core';
+import {
+  AbstractControl, FormControl, FormGroup, ValidatorFn, Validators
+} from '@angular/forms';
+import { Router } from '@angular/router';
+import { User } from 'src/app/models/shared/user.model';
+import { UserChangePassword } from 'src/app/models/user/user-change-password';
+
+@Component({
+  selector: 'app-update-password-form',
+  templateUrl: './update-password-form.component.html',
+  styleUrls: ['./update-password-form.component.sass']
+})
+export class UpdatePasswordFormComponent implements OnInit {
+  form!: FormGroup;
+  user: User;
+
+  @Output()
+  updatedUserPassword = new EventEmitter<UserChangePassword>();
+
+  constructor(private _router: Router) { }
+
+  ngOnInit() {
+    this.form = new FormGroup({
+      currentPassword: new FormControl('', Validators.required),
+      newPassword: new FormControl('', [
+        Validators.required,
+        Validators.pattern('^(?=.*[A-Za-z])(?=.*[0-9@$!%*#?&-_])[A-Za-z0-9@$!%*#?&-_]{6,20}$')
+      ]),
+      newPasswordConfirmation: new FormControl('', [
+        Validators.required,
+        Validators.pattern('^(?=.*[A-Za-z])(?=.*[0-9@$!%*#?&-_])[A-Za-z0-9@$!%*#?&-_]{6,20}$')
+      ])
```

Comment:
Firstly, check all comments from the previous Pull request, because here we have almost the same changes/code comparing with develop branch.

Feature/add page user profile editing

Reply:
added changes required in the review